lms/djangoapps/oauth2_handler/handlers.py

```diff
@@ -190,23 +189,29 @@ def find_courses(self, user, access_type, values=None):
 
         return course_ids
 
-    # pylint: disable=missing-docstring
     def _get_courses_with_access_type(self, user, access_type):
+        """
+        Returns all courses for global staff.  Otherwise, returns list of courses
+        based on role access (e.g. courses that course staff has access to).
+        """
+
         # Check the application cache and update if not present. The application
         # cache is useful since there are calls to different endpoints in close
         # succession, for example the id_token and user_info endpoints.
-
         key = '-'.join([str(self.__class__), str(user.id), access_type])
         course_ids = cache.get(key)
 
         if not course_ids:
-            courses = _get_all_courses()
-
-            # Global staff have access to all courses. Filter courses for non-global staff.
-            if not GlobalStaff().has_user(user):
-                courses = [course for course in courses if has_access(user, access_type, course)]
 
-            course_ids = [unicode(course.id) for course in courses]
+            if GlobalStaff().has_user(user):
+                # Use mongo for global staff because not all courses have roles and would thus
```

Comment:
Nit: Please update this comment, so: 
1. Indicate a TODO that this code should be optimized in the future even for global staff by caching the list of all courses in the system.
2. What does "not all courses have roles" mean?   Do you mean "Use mongo for global staff to obtain all the courses in the system"?